custom-templates/priv_instances_public_nlb_ssl/neo4j-nlb-priv-ssl.template.yaml

```diff
@@ -0,0 +1,857 @@
+            - "    coreMembers=$(echo ${coreMembers} | sed 's/ /:5000,/g')\n"
+            - "    coreMembers=$(echo \"${coreMembers}\"):5000\n"
+            - "    sed -i s/#dbms.cluster.discovery.endpoints=localhost:5000,localhost:5001,localhost:5002/dbms.cluster.discovery.endpoints=${coreMembers}/g /etc/neo4j/neo4j.conf\n"
+            - "  fi\n"
+            - "}\n"
+
+            - "start_neo4j() {\n"
+            - "  echo \"Starting Neo4j...\"\n"
+            - "  service neo4j start\n"
+            - "  neo4j-admin dbms set-initial-password \"${password}\"\n"
+            - "  while [[ \"$(curl -s -o /dev/null -m 3 -L -w '%{http_code}' http://localhost:7474 )\" != \"200\" ]];\n"
+            - "    do echo \"Waiting for cluster to start\"\n"
+            - "    sleep 5\n"
+            - "  done\n"
+            - "  echo \"Waiting another 2 minutes to give the TGs a chance to go green...\"\n"
+            - "  sleep 120\n"
+            - "  /opt/aws/bin/cfn-signal -e $? --stack \"${stackName}\" --resource Neo4jAutoScalingGroup --region \"${region}\"\n"
+            - "}\n"
+
+            - "install_neo4j_from_yum\n"
+            - "install_apoc_plugin\n"
+            - "extension_config\n"
+            - "build_neo4j_conf_file\n"
+            - "configure_graph_data_science\n"
+            - "configure_bloom\n"
+            - "start_neo4j\n"
+
+  Neo4jHTTPTargetGroup:
+    Type: AWS::ElasticLoadBalancingV2::TargetGroup
+    Properties:
+      Name: !Join
+        - '-'
+        - - !Ref 'AWS::StackName'
+          - 'http'
+          - 'tg'
+      Port: 7474
+      Protocol: TCP
+      HealthCheckIntervalSeconds: 10
+      Tags:
+        - Key: StackID
+          Value: !Ref 'AWS::StackId'
+        - Key: Name
+          Value: !Ref 'AWS::StackName'
+      VpcId: !Ref Neo4jVPC
+
+  Neo4jBoltTargetGroup:
+    Type: AWS::ElasticLoadBalancingV2::TargetGroup
+    Properties:
+      Name: !Join
+        - '-'
+        - - !Ref 'AWS::StackName'
+          - 'bolt'
+          - 'tg'
+      Port: 7687
+      Protocol: TCP
+      HealthCheckIntervalSeconds: 10
+      Tags:
+        - Key: StackID
+          Value: !Ref 'AWS::StackId'
+        - Key: Name
+          Value: !Ref 'AWS::StackName'
+      VpcId: !Ref Neo4jVPC
+
+  Neo4jNetworkLoadBalancer:
+    Type: AWS::ElasticLoadBalancingV2::LoadBalancer
+    Properties:
+      IpAddressType: ipv4
+      Name: !Join
+        - '-'
+        - - !Ref 'AWS::StackName'
+          - 'nlb'
+      Scheme: internet-facing
+      Subnets: 
+        - !Ref Neo4jPublicSubnet1
+        - !Ref Neo4jPublicSubnet2
+        - !Ref Neo4jPublicSubnet3
```

Comment:
So I think this will create a load balancer node in each AZ. From the following [page](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-elasticloadbalancingv2-loadbalancer.html#cfn-elasticloadbalancingv2-loadbalancer-subnets):

`[Network Load Balancers] You can specify subnets from one or more Availability Zones.`

Given that you have cross zone load balancing enabled, the load balancer node in the defined subnet will route to neo4j instance in all three AZs. I wonder if this should be set to one subnet, such as `Neo4jPublicSubnet2` and then the diagram can be updated to place the load balancer into Public Subnet 2?